### PR TITLE
[Feature] Add cast varchar to bitmap

### PR DESCRIPTION
**Pull request by Pslydhh**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## TARGET
Add cast varchar to bitmap.
**varchar should be serialized bytes from bitmap**.
**We just use this expr for serialized bitmap.**

## IMPLEMENTATION
- Add deserialize with **parameter max_bytes** to read from serialized datas safely.
- Internal logic of Roaring64Map is re-implemented with check for invalid bitmap.
- We check there is valid bitmap(**read_roaring_check**) before read from serialized datas.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Type.java|1|2|50.00%|[968]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Type.java|1|2|50.00%|[968]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Type.java|1|2|50.00%|[968]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Type.java|1|2|50.00%|[968]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Type.java|1|2|50.00%|[968]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Type.java|1|2|50.00%|[968]|

File: be/src/exprs/vectorized/cast_expr.cpp [hunk 1]
```diff
@@ -347,6 +348,33 @@ static ColumnPtr cast_from_string_to_hll_fn(ColumnPtr& column) {
...
+    for (int row = 0; row < viewer.size(); ++row) {
+        if (viewer.is_null(row)) {
+            builder.append_null();
+            continue;
+        }
+
+        auto value = viewer.value(row);
+
+        BitmapValue bitmap;
+        if (bitmap.deserialize(value.data, value.size)) {
```
Comment (kangkaisen) [inline be/src/exprs/vectorized/cast_expr.cpp:364]: why not use `roaring_bitmap_portable_deserialize_size` to fast check

File: be/src/types/bitmap_value.cpp [hunk 2]
```diff
@@ -743,6 +745,89 @@ bool BitmapValue::deserialize(const char* src) {
...
+            break;
+        }
+        default:
+            return false;
+        }
+        return true;
+    }
+}
+
+bool BitmapValue::split_as_uint64_t(const Slice& slice, std::vector<uint64_t>* result) {
```
Comment (kangkaisen) [inline be/src/types/bitmap_value.cpp:823]: shouldn't in BitmapValue class.

File: be/src/exprs/vectorized/cast_expr.cpp (hunk 1 above)
Comment (Pslydhh) [reply be/src/exprs/vectorized/cast_expr.cpp:364]: > why not use `roaring_bitmap_portable_deserialize_size` to fast check

ok, I will use it to remove allocate memory dynamically.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Type.java|1|2|50.00%|[968]|

**Comment by Pslydhh** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Type.java|2|2|100.00%|[]|

File: be/src/types/bitmap_value.h [hunk 3]
```diff
@@ -113,6 +114,8 @@ class BitmapValue {
     // Deserialize a bitmap value from `src`.
     // Return false if `src` begins with unknown type code, true otherwise.
     bool deserialize(const char* src);
+    // Use max_bytes to read from src safely.
+    bool deserialize(const char* src, size_t max_bytes);
```
Comment (kangkaisen) [inline be/src/types/bitmap_value.h:118]:
```suggestion
    bool valid_and_deserialize(const char* src, size_t max_bytes);
```

**Comment by Pslydhh** [comment]
run starrocks_fe_unittest

**Review by kangkaisen — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Pull request merged**